src/main/java/org/openrewrite/staticanalysis/InstanceOfPatternMatch.java

```diff
@@ -364,9 +364,7 @@ public String variableName(@Nullable JavaType type) {
                 return name;
             } else if (type instanceof JavaType.FullyQualified) {
                 String className = ((JavaType.FullyQualified) type).getClassName();
-                if (className.indexOf('.') > 0) {
-                    className = className.substring(className.lastIndexOf('.'));
-                }
+                className = className.substring(className.lastIndexOf('.') + 1);
```

Comment:
Subtle using last index of being -1 here and adding 1; unless it's found in which case you skip. Smart. :)

Reply:
Thank you!